### ISSUE TITLE
Adversarial Robustness: problem with loading pertained model

### ISSUE THREAD

**Issue by YiDongOuYang** [deepmind/deepmind-research#317]
https://github.com/deepmind/deepmind-research/blob/fba48d1e44d86628b65a31549560b7be2a25d823/adversarial_robustness/jax/eval.py#L89

Dear authors, thank you for your great work! However, I cannot load the pertained model provided by you, i.e., cifar10_linf_wrn28-10_cutmix_ddpm_v2.npy. The key in this npy file is different from the WideResNet class in model_zoo.py.

what the npy file saved:
![what the npy file saved](https://user-images.githubusercontent.com/58366413/148187876-df24c5ee-3c4b-4fde-a460-ce9aba9c8dd6.png)

what we need for params:
![what we need for params](https://user-images.githubusercontent.com/58366413/148188055-bad6bde4-c741-4be6-aa27-497631e73756.png)

what we need for state:
![what we need for state](https://user-images.githubusercontent.com/58366413/148188035-e621e473-e207-4d92-bc96-ad5af7ab63e8.png)

**Comment by sgowal** [comment]
Thank you for reporting the error. Indeed, some JAX checkpoint were wrongly imported (all PyTorch checkpoints should work). Could you try the new [cifar10_linf_wrn28-10_cutmix_ddpm_v2.npy](https://storage.googleapis.com/dm-adversarial-robustness/cifar10_linf_wrn28-10_cutmix_ddpm_v2.npy) model and let us know if it works.

**Comment by YiDongOuYang** [comment]
Dear Gowal, thank you very much for your precise and swift reply. I really appreciate it!
[https://storage.googleapis.com/dm-adversarial-robustness/cifar10_linf_wrn28-10_cutmix_ddpm_v2.npy](url) This model works well!

BTW, the PyTorch checkpoints work very well! However, there is no training code provided in this repository (pytorch version), and we can only use the eval.py to test the clear accuracy rather than robust accuracy. Even I get the pretrained model, I cannot finetune the model or test the model under robust criterion unless I write these codes in pytorch. 

I do have a look at Rahul Rade's implementation [https://github.com/imrahulr/adversarial_robustness_pytorch](url). However, there is also some gaps, i.e., they don't provide pre-trained model, the pre-trained model provided by you cannot load into his repository. (I believe it can with some small efforts.)

I am wondering could you please release some of the necessary code of pytorch version? (I don't want to impose on you. If it is not convenient, I could also fill in the gap between your pretrained model and Rahul Rade's implementation to help others like me to adopt your code!) Thank you again:D

**Comment by sgowal** [comment]
Hi YiDongOuYang,

Unfortunately we do not plan to release an official PyTorch version of the training pipeline. We encourage you to use the JAX implementation if you can.

Feel free to reach out to me directly if you find other JAX checkpoints that cannot be loaded. I apologize for the inconvenience.

Regards,
Sven

**Comment by YiDongOuYang** [comment]
Hi Sven,

No worry! Thank you again! 

I check other JAX checkpoints just now. I found that four links for cifar-10 datasets are not correct [https://github.com/deepmind/deepmind-research/tree/master/adversarial_robustness/iclrw2021doing](url).

Best wishes,
Yidong
